src/briefcase/commands/convert.py

```diff
@@ -0,0 +1,866 @@
+import re
+from difflib import SequenceMatcher
+from functools import cached_property, wraps
+from pathlib import Path
+from shutil import copy2, copytree
+from tempfile import TemporaryDirectory
+from typing import Callable, Optional
+from urllib.parse import urlparse
+
+from packaging.version import Version
+
+from ..config import is_valid_app_name
+from .base import BaseCommand
+from .new import NewCommand, titlecase
+
+try:
+    import tomllib
+except ModuleNotFoundError:
+    import tomli as tomllib
+
+import briefcase
+from briefcase.config import make_class_name
+from briefcase.exceptions import BriefcaseCommandError, TemplateUnsupportedVersion
+from briefcase.integrations import git
+
+
+def normalize_app_name(app_name):
+    """Normalize a valid PEP508 name.
+
+    https://packaging.python.org/en/latest/specifications/name-normalization/#valid-non-normalized-names
+    """
+    return re.sub(r"[-_.]+", "-", app_name).lower()
+
+
+def find_most_similar_path_by_name(paths: list[Path], name: str) -> Path:
+    """Given a list of paths and a name, find the path with the most similar name.
+
+    Ties are resolved by selecting the first entry in the paths list.
+
+    :param paths: List of paths to search in
+    :param name: Name to compare the path names with
+    """
+    if not paths:
+        raise ValueError("Cannot find most similar path with no paths.")
+
+    max_similarity = float("-inf")
+
+    for path in paths:
+        similarity = SequenceMatcher(None, name, path.name).ratio()
+        if similarity > max_similarity:
+            max_similarity, most_similar = similarity, path
+
+    return most_similar
+
+
+class ConvertCommand(BaseCommand):
+    cmd_line = "briefcase convert"
+    command = "convert"
+    platform = "all"
+    output_format = None
+    description = "Set up an existing project for Briefcase."
+
+    def bundle_path(self, app):
+        """A placeholder; Convert command doesn't have a bundle path."""
+        raise NotImplementedError()
+
+    def binary_path(self, app):
+        """A placeholder; Convert command doesn't have a binary path."""
+        raise NotImplementedError()
+
+    def parse_config(self, filename):
+        """There is no configuration when setting up existing project; this
+        implementation overrides the base so that no config is parsed."""
+        pass
+
+    def verify_tools(self):
+        """Verify that the tools needed to run this command exist.
+
+        Raises MissingToolException if a required system tool is missing.
+        """
+        super().verify_tools()
+        git.verify_git_is_installed(tools=self.tools)
+
+    @wraps(NewCommand.add_options)
+    def add_options(self, parser):
+        return NewCommand.add_options(self, parser)
```

Comment:
This pattern feels like it's trying to implement subclassing without actually doing subclassing. AFAICT, if ConvertCommand is a subclass of NewCommand (or vice versa), most of this implementation would no longer be required.

Reply:
We were considering that. However, we weren't sure if it made sence to say that a `ConvertCommand` is a special case of a `NewCommand` (or vice versa). One possible solution is to abstract the shared behaviour into a `SetupProjectCommand` that both `ConvertCommand` and `NewCommand` inherits from. What do you think of that solution?

Reply:
Based on some of the subsequent logic, my initial impression is that "Convert" is a subclass of "New" - after all, Convert will be invoking the same "ask questions and roll out a template" logic, but with an additional "merge" step, and some variations on which questions are asked.

However, a common base class also seems like it could be a viable option. It's difficult to say for certain which would be better without having worked through all the details myself.

Reply:
We opted for subclassing, and it seemed to work nicely.